llvm/lib/CodeGen/AtomicExpandPass.cpp

```diff
@@ -1866,11 +1844,22 @@ bool AtomicExpand::expandAtomicOpToLibcall(
   FunctionType *FnType = FunctionType::get(ResultTy, ArgTys, false);
   // XXXAR: ugly hack to reduce size of the diff: for capability types we set
   // RTLibType to the generic one and then manually append _cap instead of
-  // adding the 20+ new entries to RuntimeLibcalls.def
+  // adding the 20+ new entries to RuntimeLibcalls.def. We also suffix with
+  // _c for capability pointer arguments in hybrid mode.
+  assert(TLI->getLibcallName(RTLibType));
```

Comment:
Hm, this was already checked on line 1740 (now 1705), this really should be very dead, so much so that due to `getLibcallName` being const it should be able to be optimised out.

Reply:
I added that assert while debugging the amdgpu crash, but forgot to delete it after cherry picking the upstreamed fix.